fs/inode/fs_files.c

```diff
@@ -96,30 +96,18 @@ static int files_extend(FAR struct filelist *list, size_t row)
 
   orig_rows = list->fl_rows;
 
-  files = kmm_malloc(sizeof(FAR struct file *) * row);
-  DEBUGASSERT(files);
+  block_size = sizeof(struct file) * CONFIG_NFILE_DESCRIPTORS_PER_BLOCK;
+
+  files = kmm_zalloc((sizeof(FAR struct file *) + block_size) * row);
```

Comment:
Normally CONFIG_NFILE_DESCRIPTORS_PER_BLOCK is 6 for most of the boards, to it will allocate: (size of file * 7) * row. So I think it will allocate a considerable amount of memory depending on the number of row.

Note: it is 7 instead of 6 because you are sum up +1 sizeof() inside kmm_zalloc()

Reply:
we can change the row array to hold `struct file` instead `struct file *`, which could save `block_size * sizeof(void *)` per row. From the origin patch: https://github.com/apache/nuttx/pull/3045/files#diff-285d196ad580d55fdaef3fbe54702ee3ddeed0d93cba52a5441d1866e0031726R1061
the default setting consumes smaller memory than before even this patch allocate `struct file` directly.